src/test/java/com/jwplayer/southpaw/util/ByteArraySetTest.java

```diff
@@ -46,6 +49,1682 @@ public ByteArraySet createSmallSet() {
+        }
+    }
+
+    @Test
+    public void testMergedSimilarRemove() {
+        testSimilarRemove(true);
+    }
+
+    @Test
+    public void testUnmergedSimilarRemove() {
+        testSimilarRemove(false);
+    }
+
+    public void testToArray(int size) {
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+        }
+        if (size > 0) {
+            assertFalse(set.isEmpty());
+        } else {
+            assertTrue(set.isEmpty());
+        }
+
+        List<ByteArray> alreadySeenVals = new ArrayList<ByteArray>();
+        int seenCount = 0;
+        for(ByteArray val: set.toArray()) {
+            assertFalse(alreadySeenVals.contains(val));
+            alreadySeenVals.add(val);
+            assertTrue(vals.contains(val));
+            seenCount++;
+        }
+        assertEquals(size, seenCount);
+    }
+
+    @Test
+    public void emptyToArray() {
+        testToArray(0);
+    }
+
+    @Test
+    public void reallySmallToArray() {
+        testToArray(1);
+    }
+
+    @Test
+    public void smallToArray() {
+        testToArray(2);
+    }
+
+    @Test
+    public void regularToArray() {
+        testToArray(150);
+    }
+
+    @Test
+    public void bigToArray() {
+        testToArray(750);
+    }
+
+    @Test
+    public void reallyBigToArray() {
+        testToArray(3567);
+    }
+
+    public void testContains(int size) {
+        ByteArraySet set = new ByteArraySet();
+
+        List<ByteArray> vals = getRandomByteArrays(size);
+
+        for (ByteArray val: vals) {
+            assertTrue(set.add(val));
+        }
+        if (size > 0) {
+            assertFalse(set.isEmpty());
+        } else {
+            assertTrue(set.isEmpty());
+        }
+        Collections.shuffle(vals);
+
+        for(ByteArray val: vals) {
+            assertTrue(set.contains(val));
+        }
+
+        List<ByteArray> absentVals = getRandomByteArrays(10);
+        for(ByteArray val: absentVals) {
+            if (vals.contains(val)) {
+                assertTrue(set.contains(val));
+            } else {
+                assertFalse(set.contains(val));
+            }
+        }
+    }
+
+    @Test
+    public void emptyContains() {
+        testContains(0);
+    }
+
+    @Test
+    public void reallySmallContains() {
+        testContains(1);
+    }
+
+    @Test
+    public void smallContains() {
+        testContains(2);
+    }
+
+    @Test
+    public void regularContains() {
+        testContains(150);
+    }
+
+    @Test
+    public void bigContains() {
+        testContains(750);
+    }
+
+    @Test
+    public void reallyBigContains() {
+        testContains(3567);
+    }
+
+    @Test
+    public void testEmptyLastValueChunkIteratorBug() {
+        ByteArraySet set = new ByteArraySet();
+
+        List<String> stringVals = new ArrayList<String>();
```

Comment:
I'd suggest we use a function here to generate a random alphamumeric string. Then we can loop through and build this array dynamically.

Reply:
The issue is we need to selectively some of these values for the bug to appear. I can look at how we could make this test shorter.